### PR TITLE
Image Editor: add action to reset all state (inc src, fileName), use it to reset image on Cancel click

### PR DESCRIPTION
**Pull request by lamosty**
Fixes a regression which was introduced in #8408. After editing an image in image editor, then closing image editor by clicking Cancel and then editing the same image, only placeholders would be visible and no image.

In this PR, we add new action type and action creator to reset all of image's state including `src`, `fileName`, `mimeType` and `title`.

## Testing

1. Open image for editing and click "Cancel";
2. Open the same image for editing: does it load?
3. Open image for editing and click "Done";
4. While copy of the image is being uploaded, open the original image for editing again: does it load?
5. Try to do the above in a quick way: does it still work?

## Todo

- [x] Add tests
- [x] Dispatch this action on pressing the `ESC` key as well
- [x] Dispatch this action in the `onImageExtracted` handler to reset the image after clicking on "Done"

/cc @gwwar @hoverduck

### CODE REVIEW

**Comment by hoverduck** [comment]
Tested, looks good to me

File: client/state/ui/editor/image-editor/test/reducer.js [hunk 1]
```diff
@@ -77,6 +78,14 @@ describe( 'reducer', () => {
 
 			expect( state ).to.be.false;
 		} );
+
+		it( 'should change to false on reset all', () => {
```
Comment (gwwar) [inline client/state/ui/editor/image-editor/test/reducer.js]: How about `Should have no changes after reset all`?

Comment (gwwar) [inline client/state/ui/editor/image-editor/test/reducer.js]: Oh I suppose the describe block above might make this readable enough. 👍 So feel free to ignore the above comment

**Comment by gwwar** [comment]
:shipit:

**Comment by lamosty** [comment]
Thanks for the reviews @hoverduck and @gwwar!

**Pull request merged**